Scripts/Handles/BaseHandle.cs

```diff
@@ -77,23 +90,32 @@ protected virtual HandleEventData GetHandleEventData(RayEventData eventData)
 			return new HandleEventData(eventData.rayOrigin, UIUtils.IsDirectEvent(eventData));
 		}
 
+		public int IndexOfHoverSource(Transform rayOrigin)
+		{
+			return m_HoverSources.IndexOf(rayOrigin);
+		}
+
+		public int IndexOfDragSource(Transform rayOrigin)
+		{
+			return m_DragSources.IndexOf(rayOrigin);
+		}
+
 		public void OnBeginDrag(RayEventData eventData)
 		{
 			if (!UIUtils.IsValidEvent(eventData, selectionFlags))
 				return;
 
-			m_DragSources.Add(eventData.rayOrigin);
-			startDragPosition = eventData.pointerCurrentRaycast.worldPosition;
+			var rayOrigin = eventData.rayOrigin;
+			m_DragSources.Add(rayOrigin);
+			startDragPositions[rayOrigin] = eventData.pointerCurrentRaycast.worldPosition;
 
 			var handleEventData = GetHandleEventData(eventData);
 
 			//Double-click logic
-			var timeSinceLastClick = (float) (DateTime.Now - m_LastClickTime).TotalSeconds;
+			var timeSinceLastClick = (float)(DateTime.Now - m_LastClickTime).TotalSeconds;
 			m_LastClickTime = DateTime.Now;
```

Comment:
Hm, I suppose you're right. Seems like a pretty slim chance of happening, but we might as well be consistent.

Now that we're on the subject, we should remember to check the double click behavior when moving to the new input system. Currently, it is possible to double click so fast that we dont catch the return to a zero-value, and essentially you can double click between frames. If we send input events more quickly than the update rate, we can probably fix this issue.